File: app/src/main/java/com/example/month6/view/MainActivity.java
package com.example.month6.view;

import android.os.Bundle;
import android.widget.RadioGroup;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;

import com.example.month6.view.mainactivity_frag.HomeFrag;
import com.example.month6.view.mainactivity_frag.ShowFrag;
import com.example.month6.view.mainactivity_frag.MoneyFrag;
import com.example.month6.view.mainactivity_frag.MoreFrag;
import com.example.month6.R;

import java.util.ArrayList;

public class MainActivity extends AppCompatActivity {

    private RadioGroup radioGroup;
    ArrayList<Fragment> list=new ArrayList<>();

    HomeFrag f1=new HomeFrag(this);
    ShowFrag f2=new ShowFrag();
    MoneyFrag f3=new MoneyFrag();
    MoreFrag f4=new MoreFrag();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        initView();
        showFragment(f1);
    }

    private void initView() {
        radioGroup = findViewById(R.id.radioGroup);
        //底部监听
        radioGroup.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {
                switch (checkedId){
                    case R.id.radioBut1: showFragment(f1);break;
                    case R.id.radioBut2: showFragment(f2);break;
                    case R.id.radioBut3: showFragment(f3);break;
                    case R.id.radioBut4: showFragment(f4);break;
                }
            }
        });
    }

    //加载fragment
    private void showFragment(Fragment frag){
        if (!frag.isAdded()){
            getSupportFragmentManager().beginTransaction().add(R.id.frameLayout,frag).commit();
            list.add(frag);
        }
        for (Fragment i : list){
            getSupportFragmentManager().beginTransaction().hide(i).commit();
        }
        getSupportFragmentManager().beginTransaction().show(frag).commit();
    }
}

File: app/src/main/java/com/example/p2pfinancial/MainActivity.kt
package com.example.p2pfinancial

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import com.example.p2pfinancial.fragment.Fragment1
import com.example.p2pfinancial.fragment.Fragment2
import com.example.p2pfinancial.fragment.Fragment3
import com.example.p2pfinancial.fragment.Fragment4
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    var fragList = listOf(Fragment1(), Fragment2(), Fragment3(), Fragment4())
    var currentFragment: Fragment? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        initBottomBar()

    }

    private fun initBottomBar() {
        bottombar.setOneCliecked(1)
        getshowFrag(fragList[0])
        bottombar.setOnTapListener {
            when (it) {
                0 -> {
                    getshowFrag(fragList[0])
                }
                1 -> {
                    getshowFrag(fragList[1])
                }
                2 -> {
                    getshowFrag(fragList[2])
                }
                3 -> {
                    getshowFrag(fragList[3])
                }
            }
        }
        bottombar.setTapNum(4)
        bottombar.setTapText(arrayOf("首页", "投资", "我的资金", "更多"))
        bottombar.setTapDrables(
            arrayOf(
                resources.getDrawable(R.drawable.rb_main_selector),
                resources.getDrawable(R.drawable.rb_invest_selector),
                resources.getDrawable(R.drawable.rb_myinvest_selector),
                resources.getDrawable(R.drawable.rb_more_selector)
            )
        )
    }
    fun getshowFrag(fragment: Fragment) {
        val beginTransaction = supportFragmentManager.beginTransaction()
        if (currentFragment != null) {
            beginTransaction.hide(currentFragment!!)
        }
        if (fragment.isAdded) {
            beginTransaction.show(fragment)
        } else {
            beginTransaction.add(R.id.frag, fragment)
        }
        beginTransaction.commit()
        currentFragment = fragment
    }
}
